tfjs-layers/src/layers/preprocessing/random_width.ts

```diff
@@ -0,0 +1,160 @@
+ */
+
+const INTERPOLATION_KEYS = ['bilinear', 'nearest'] as const;
+export const INTERPOLATION_METHODS = new Set(INTERPOLATION_KEYS);
+type InterpolationType = typeof INTERPOLATION_KEYS[number];
+
+export class RandomWidth extends BaseRandomLayer {
+  /** @nocollapse */
+  static override className = 'RandomWidth';
+  private readonly factor: number | [number, number];
+  private readonly interpolation?: InterpolationType;  // defualt = 'bilinear
+  private readonly seed?: number; // default null
+  private widthLower: number;
+  private widthUpper: number;
+  private imgHeight: number;
+  private adjustedWidth: number;
+  private widthFactor: Tensor<Rank.R3>|Tensor<Rank.R4>;
+
+  constructor(args: RandomWidthArgs) {
+    super(args);
+    this.factor = args.factor;
+
+    if (Array.isArray(this.factor) && this.factor.length === 2) {
+      this.widthLower = this.factor[0];
+      this.widthUpper = this.factor[1];
+    } else if (!Array.isArray(this.factor) && this.factor > 0){
+      this.widthLower = -this.factor;
+      this.widthUpper = this.factor;
+    } else {
+      throw new ValueError(
+        `Invalid factor: ${this.factor}. Must be positive number or tuple of 2 numbers`
+      );
+    }
+    if (this.widthLower < -1.0 || this.widthUpper < -1.0) {
+      throw new ValueError(
+        `factor must have values larger than -1. Got: ${this.factor}`
+      );
+    }
+
+    if (this.widthUpper < this.widthLower) {
+      throw new ValueError(
+        `factor cannot have upper bound less than lower bound.
+        Got upper bound: ${this.widthUpper}.
+        Got lower bound: ${this.widthLower}
+      `);
+    }
+
+    if(args.seed) {
+      this.seed = args.seed;
+    } else {
+      this.seed = null;
+    }
+
+    if (args.interpolation) {
+      if (INTERPOLATION_METHODS.has(args.interpolation)) {
+        this.interpolation = args.interpolation;
+      } else {
+        throw new ValueError(`Invalid interpolation parameter: ${
+            args.interpolation} is not implemented`);
+      }
+    } else {
+      this.interpolation = 'bilinear';
+    }
+  }
+
+  override getConfig(): serialization.ConfigDict {
+    const config: serialization.ConfigDict = {
+      'factor': this.factor,
+      'interpolation': this.interpolation,
+      'seed': this.seed,
+    };
+
+    const baseConfig = super.getConfig();
+    Object.assign(config, baseConfig);
+    return config;
+  }
+
+  override computeOutputShape(inputShape: Shape|Shape[]): Shape|Shape[] {
+    inputShape = getExactlyOneShape(inputShape);
+    const numChannels = inputShape[2];
+    return [this.imgHeight, this.adjustedWidth, numChannels];
+  }
+
+  override call(inputs: Tensor<Rank.R3>|Tensor<Rank.R4>,
+    kwargs: Kwargs): Tensor[]|Tensor {
+
+    return tidy(() => {
+        const input = getExactlyOneTensor(inputs);
+        const inputShape = input.shape;
+        this.imgHeight = inputShape[inputShape.length - 3];
+        const imgWidth = inputShape[inputShape.length - 2];
```

Comment:
Minor nit: You can use `Array.at()` to get python-style indexing from the end of the array.
```suggestion
        this.imgHeight = inputShape.at(-3);
        const imgWidth = inputShape.at(-2);
```